types/lib/model.d.ts

```diff
@@ -1671,7 +1671,21 @@ export abstract class Model<T = any, T2 = any> extends Hooks {
     options?: string | ScopeOptions | (string | ScopeOptions)[] | WhereAttributeHash
   ): M;
 
+  /**
+   * Add a new scope to the model
+   *
+   * This is especially useful for adding scopes with includes, when the model you want to
+   * include is not available at the time this model is defined. By default this will throw an
+   * error if a scope with that name already exists. Pass `override: true` in the options
+   * object to silence this error.
+   *
+   * @param name The name of the scope to add
+   * @param scope An object of `FindOptions`, or a function that returns one
+   * @param options An object of options for creating the scope
+   * @param options.override Should existing scopes of the same name be overridden? Default: `false`
+   */
   public static addScope(name: string, scope: FindOptions, options?: AddScopeOptions): void;
+  public static addScope(name: string, scope: (...args: any[]) => FindOptions, options?: AddScopeOptions): void;
```

Comment:
@SimonSchick `unknown[]` will not work in this case, because arguments typed at the callsite will not be assignable to `unknown`. For example, with the test I've added:

```tsx
User.addScope(
  'withFirstName',
  (firstName: string) => ({
    where: { firstName },
  }),
);
```

…the type `...args: unknown[]` will raise the following error:

<img width="680" alt="Screenshot 2019-04-27 10 56 55" src="https://user-images.githubusercontent.com/744542/56851316-977a1e00-68db-11e9-8032-a66cca7e5564.png">

Reply:
I figured TS would treat explicitly declaring the variables as a valid cast, guess it doesn't 😅